src/media_changer.py

```diff
@@ -3702,7 +3702,1513 @@ def DoWork(self, function, ticket):
+        # do the work ...
+
+        # ... if this is a mount, dismount first
+        if ticket['function'] == "mount":
+	    message = "mcDoWork> child prepare dismount for %s" % \
+			(common_message,)
+            Trace.trace(ACTION_LOG_LEVEL, message)
+
+	    # don't print a failure  (no tape mounted) message that is
+	    # really a success
+            self.logdetail = 0
+	    # To Do: The downside to passing the volume we want to mount is
+	    # that there is a warning message about the MC expecting to
+	    # unmount that tape.  Some blank value should be passed to
+	    # avoid that false warning.
+	    dismount_ticket = {'work'           : 'unloadvol',
+			       'vol_ticket'     : ticket['vol_ticket'],
+			       'drive_id'       : ticket['drive_id'],
+			       }
+	    sts = self.prepare(dismount_ticket)
+            self.logdetail = 1 # back on
+
+	    message = "%s returned %s" % (message, sts[2])
+            Trace.trace(ACTION_LOG_LEVEL, message)
+
+	    # XXX - Why isn't sts processed for errors here?
+
+	message = "mcDoWork> child doing %s" % (common_message,)
+	Trace.trace(ACTION_LOG_LEVEL, message)
+
+	sts = function(ticket) #Call the function!
+
+	message = "mcDoWork> child %s returned %s" % (common_message, sts)
+	Trace.trace(ACTION_LOG_LEVEL, message)
+
+        ticket["status"] = sts
+        # Send status back to MC parent via pipe then via dispatching_worker
+	# and WorkDone ticket so dispatching_worker calls WorkDone().
+	ticket['work'] = "WorkDone"
+
+        #There must be a better way to write to the pipe connected to the
+	# parent process.  Probably with callback.py.
+	msg = repr(('0','0',ticket))
+	bytecount = "%08d" % (len(msg),)
+	try:
+		os.write(comm_pipe[1],  bytecount)
+		os.write(comm_pipe[1],  msg)
+		os.close(comm_pipe[1])
+	except (OSError, IOError), msg:
+		message = "mcDoWork> child %s failed reporting to parent: %s" \
+			  % (common_message, str(msg))
+		Trace.log(e_errors.ERROR, message)
+	except:
+		Trace.handle_error()
+
+        return
+
+    def WorkDone(self, ticket):
+        # remove work from outstanding work list
+        self.remove_from_work_list(ticket)
+        # log what was done
+        status = ticket.get('status', None)
+        if status and e_errors.is_ok(status[0]):
+            level = e_errors.INFO
+        else:
+            level = e_errors.ERROR
+        common_message = self.get_common_message_string(ticket)
+        Trace.log(level, "FINISHED %s returned %s" % (common_message, status))
+        # log the new work list
+        self.log_work_list(ticket)
+        if not ticket.get('no_reply', None):
+            self.reply_to_caller(ticket)
+	try:
+		os.close(self.p[1])
+	except:
+		exc, msg, tb = sys.exc_info()
+		Trace.log(e_errors.ERROR, "WorkDone failed:  %s %s %s"% (exc, msg, traceback.format_tb(tb)))
+		del tb
+
+    def __init__(self, medch, max_work=1, csc=None):
+
+        MediaLoaderMethods.__init__(self,medch,max_work,csc)
```

Comment:
Looks like misaligned indent

Reply:
It does not look to me. Why do you think so?